aviary/subsystems/propulsion/hamilton_standard.py

```diff
@@ -0,0 +1,1021 @@
+    .675, .685, .700, .735, .777, .810, .845, .870, .890, .905, .920, .930,  # X = 0.15
+    .525, .540, .565, .615, .670, .710, .745, .790, .825, .860, .880, .895,  # X = 0.20
+    .225, .260, .320, .375, .430, .495, .550, .610, .660, .710, .740, .775,  # X = 0.30
+])
+
+
+class PreHamiltonStandard(om.ExplicitComponent):
+    """
+    Pre-process parameters needed by HamiltonStandard component
+    """
+
+    def initialize(self):
+        self.options.declare('num_nodes', types=int)
+
+    def setup(self):
+        nn = self.options['num_nodes']
+
+        self.add_input(Aircraft.Engine.PROPELLER_DIAMETER, val=0.0, units='ft')
+        self.add_input(Dynamic.Mission.PROPELLER_TIP_SPEED,
+                       val=np.zeros(nn), units='ft/s')
+        self.add_input(Dynamic.Mission.SHAFT_POWER, val=np.zeros(nn), units='hp')
+        self.add_input(Dynamic.Mission.DENSITY, val=np.zeros(nn), units='slug/ft**3')
+        self.add_input(Dynamic.Mission.VELOCITY, val=np.zeros(nn), units='knot')
+        self.add_input(Dynamic.Mission.TEMPERATURE, val=np.zeros(nn), units='degR')
+
+        self.add_output('power_coefficient', val=np.zeros(nn), units='unitless')
+        self.add_output('adv_ratio', val=np.zeros(nn), units='unitless')
+        self.add_output('mach', val=np.zeros(nn), units='unitless')
+        self.add_output('tip_mach', val=np.zeros(nn), units='unitless')
+        self.add_output('density_ratio', val=np.zeros(nn), units='unitless')
+
+    def setup_partials(self):
+        arange = np.arange(self.options['num_nodes'])
+
+        self.declare_partials(
+            'density_ratio', Dynamic.Mission.DENSITY, rows=arange, cols=arange)
+        self.declare_partials('tip_mach', [
+            Dynamic.Mission.PROPELLER_TIP_SPEED,
+            Dynamic.Mission.TEMPERATURE,
+        ], rows=arange, cols=arange)
+        self.declare_partials('mach', [
+            Dynamic.Mission.VELOCITY,
+            Dynamic.Mission.TEMPERATURE,
+        ], rows=arange, cols=arange)
+        self.declare_partials('adv_ratio', [
+            Dynamic.Mission.VELOCITY,
+            Dynamic.Mission.PROPELLER_TIP_SPEED,
+        ], rows=arange, cols=arange)
+        self.declare_partials('power_coefficient', [
+            Dynamic.Mission.SHAFT_POWER,
+            Dynamic.Mission.DENSITY,
+            Dynamic.Mission.PROPELLER_TIP_SPEED,
+        ], rows=arange, cols=arange)
+        self.declare_partials('power_coefficient', Aircraft.Engine.PROPELLER_DIAMETER)
+
+    def compute(self, inputs, outputs):
+        outputs['density_ratio'] = inputs[Dynamic.Mission.DENSITY] / RHO_SEA_LEVEL_ENGLISH
+        sqrt_temp_ratio = math.sqrt(TSLS_DEGR / inputs[Dynamic.Mission.TEMPERATURE])
+        # vktas = Dynamic.Mission.VELOCITY*0.5924838739671128
+        vktas = inputs[Dynamic.Mission.VELOCITY]
+        # at sea level, 1 knot = .00150933 (actually 0.00149984 Mach)
+        outputs['mach'] = 0.00150933 * vktas * sqrt_temp_ratio
```

Comment:
should we just be using the flight_condition component?

Reply:
or, alternatively, just have mach and velocity be inputs, since I believe there is a flight_condition component further up that could provide both values if they are promoted out of HS